clang/lib/Sema/SemaTemplateInstantiateDecl.cpp

```diff
@@ -6198,7 +6210,10 @@ static void processSYCLKernel(Sema &S, FunctionDecl *FD, MangleContext &MC) {
   if (S.LangOpts.SYCLIsDevice) {
     S.ConstructOpenCLKernel(FD, MC);
   } else if (S.LangOpts.SYCLIsHost) {
-    CXXRecordDecl *CRD = (*FD->param_begin())->getType()->getAsCXXRecordDecl();
+    QualType KernelParamTy = (*FD->param_begin())->getType();
+    const CXXRecordDecl *CRD = (KernelParamTy->isReferenceType()
```

Comment:
Are these changes related to the attribute patch?

Reply:
Yes this was required to enable attribute on sycl kernel during host compilation. Since host support is not required, I think this can be reverted in next patch.

Reply:
There is an existing bug in compiler causing a crash when attributes are applied on kernel and we compile for host. I kept this change so I could include the host test. The test crashes without this change.

Reply:
Thanks for the explanation!